drivers/docker/driver.go

```diff
@@ -456,6 +462,10 @@ CREATE:
 		// There is still a very small chance this is possible even with the
 		// coordinator so retry.
 		return nil, nstructs.NewRecoverableError(createErr, true)
+	} else if isDockerTransientError(createErr) && attempted < 5 {
+		attempted++
+		time.Sleep(1 * time.Second)
+		goto CREATE
 	}
 
 	return nil, recoverableErrTimeouts(createErr)
```

Comment:
It's not obvious to me why `recoverableErrTimeout` are used.  `StartTask` doesn't special case them, and wraps them before returning.  I considered removing it completely and doing more retries locally, but decided to be conservative and double check goal before making further changes to driver.

Reply:
I don't have much context to add here other than that it is was brought over from pre-0.9 drivers:
https://github.com/hashicorp/nomad/blob/fcc4149c55399eb4979cd3fe3fb983cfe6c8449a/client/driver/docker.go#L1701

Reply:
Sounds good.  Makes sense for more aggressive refactor.